migration/manager.py

```diff
@@ -148,8 +148,11 @@ def convert_data_to_tool_tab(cls, data: dict[str, Any]) -> ExternalToolTab:
             position=data['position']
         )
 
-    def get_tool_tabs(self) -> list[ExternalToolTab]:
-        results = self.api.get_results_from_pages(f'/courses/{self.course.id}/tabs')
+    def create_course_log_message(self, message: str) -> str:
+        return f'{self.course}\n{message}\n- - -'
```

Comment:
This adds a course identifier to all log messages originating from `replace_tool_tab`. This helps you identify what actually happened, even if the ordering gets all jumbled because of the async nature of the API calls.

Reply:
Looks good!